webpack/assets/javascripts/react_app/redux/API/APIHelpers.js

```diff
@@ -0,0 +1,60 @@
+import { STATUS } from '../../constants';
+
+const getModifiedState = (
+  key,
+  state,
+  newState,
+  payload,
+  manageResponseByID
+) => {
+  const { id } = payload;
+
+  if (manageResponseByID) {
+    if (state[key]) {
+      return state.set(key, {
+        ...state[key],
+        [id]: { ...state[key][id], ...newState },
+      });
+    }
+    return state.set(key, { ...state[key], [id]: newState });
+  }
```

Comment:
Instead of this maybe you can pass in the API middleware call 
`{  
type: API_OPERATIONS.GET,
...,
formatResponse: (data, payload) => ({ [payload.id]: data })
}` 
So if the API uses a different prop that is not called id you can easily change it

Reply:
thanks, that's exactly what I was thinking, soon I will update :+1: